cmd/podman/pod_rm.go

```diff
@@ -53,20 +55,15 @@ func podRmCmd(c *cli.Context) error {
 	}
 	defer runtime.Shutdown(false)
 
-	args := c.Args()
-
-	if len(args) == 0 && !c.Bool("all") && !c.Bool("latest") {
-		return errors.Errorf("specify one or more pods to remove")
-	}
-
+	ctx := getContext()
 	var delPods []*libpod.Pod
 	var lastError error
-	if c.IsSet("all") {
+	if c.Bool("all") {
 		delPods, err = runtime.GetAllPods()
 		if err != nil {
 			return errors.Wrapf(err, "unable to get pod list")
 		}
-	} else if c.IsSet("latest") {
+	} else if c.Bool("latest") {
```

Comment:
No need for else?  You checked above correct?

Reply:
yeah that's true, though I kind of feel having an if, then if, then else looks clunky:
```
    if c.Bool("all") {                                                                                                                                                                                             
        pods, err = runtime.Pods()                                                                                                                                                                                 
        if err != nil {                                                                                                                                                                                            
            return errors.Wrapf(err, "unable to get running pods")                                                                                                                                                 
        }                                                                                                                                                                                                          
    }                                                                                                                                                                                                              
    if c.Bool("latest") {                                                                                                                                                                                          
        pod, err := runtime.GetLatestPod()                                                                                                                                                                         
        if err != nil {                                                                                                                                                                                            
            return errors.Wrapf(err, "unable to get latest pod")                                                                                                                                                   
        }                                                                                                                                                                                                          
        pods = append(pods, pod)                                                                                                                                                                                   
    } else {                                                                                                                                                                                                       
        for _, i := range args {                                                                                                                                                                                   
            pod, err := runtime.LookupPod(i)                                                                                                                                                                       
            if err != nil {                                                                                                                                                                                        
                if lastError != nil {                                                                                                                                                                              
                    fmt.Fprintln(os.Stderr, lastError)                                                                                                                                                             
                }                                                                                                                                                                                                  
                lastError = errors.Wrapf(err, "unable to find pod %s", i)                                                                                                                                          
                continue                                                                                                                                                                                           
            }                                                                                                                                                                                                      
            pods = append(pods, pod)                                                                                                                                                                               
        }                                                                                                                                                                                                          
    }  
```
up to you.

Reply:
I am saying you need no elses here at all.
You are appending the pods on .

Reply:
...long day
got it